### PR TITLE
feat: exclude non-onboarded trainees

### PR DESCRIPTION
**Pull request by Judge40**
Exclude non-onboarded trainees when assessing pilot placement inclusion, this category includes Dental, Foundation and Public Health trainees.

Placement exclusion can be based on the grade matching one of
 - `DCT1`
 - `DCT2`
 - `DCT3`
 - `DFT`
 - `F1`
 - `F2`

Programme Membership exclusion can be based on the specialty name matching one of
 - "Public Health Medicine"
 - "Foundation"

Or Programme Membership type matching one of
 - `VISITOR`
 - `LAT`

Or no curricula sub-type matching
 - `MEDICAL_CURRICULUM`
 - `MEDICAL_SPR`

TIS21-6643

### CODE REVIEW

**Comment by Judge40** [comment]
@ReubenRobertsHEE Took a lot from the existing pilot filtering and hooked it up as an "onboardable" check. There will be a lot of repetition when the pilot checks are done but I wanted to keep these checks clear/separated and I'm not too concerned about optimisation at this point.

The tests might give a better insight in to whether I've got this right than the code will do, just because it's calling existing filtering/check logic.

File: src/test/java/uk/nhs/hee/trainee/details/api/PlacementResourceTest.java [hunk 1]
```diff
@@ -259,23 +299,66 @@ void shouldThrowBadRequestWhenPlacementPilot2024Exception() throws Exception {
...
                 "40", "1")
                 .contentType(MediaType.APPLICATION_JSON))
         .andExpect(status().isOk())
-        .andExpect(content().string(String.valueOf(isPilotRollout2024)))
+        .andExpect(content().string("true"))
         .andReturn();
   }
 
+  @Test
+  void shouldReturnTrueWhenBothCanBeOnboardedAndInPilotRollout2024False() throws Exception {
```
Comment (ReubenRobertsHEE) [inline src/test/java/uk/nhs/hee/trainee/details/api/PlacementResourceTest.java:317]: shouldReturnFalse... surely?

File: src/test/java/uk/nhs/hee/trainee/details/api/PlacementResourceTest.java [hunk 2]
```diff
@@ -259,23 +299,66 @@ void shouldThrowBadRequestWhenPlacementPilot2024Exception() throws Exception {
...
+                "40", "1")
+                .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk())
+        .andExpect(content().string("false"))
+        .andReturn();
+  }
+
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  void shouldReturnTrueWhenOneOfCanBeOnboardedOrInPilotRollout2024False(boolean isValid)
```
Comment (ReubenRobertsHEE) [inline src/test/java/uk/nhs/hee/trainee/details/api/PlacementResourceTest.java:332]: same ☝️

**Review by ReubenRobertsHEE — APPROVED** [review]
LGTM, thanks Andy 🤞 😀

**Pull request merged**